app/main.py

```diff
@@ -1,2 +1,57 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.size = 8
+        self.count = 0
+        self.buckets = [[] for _ in range(self.size)]
+
+    def _hash(self, key: Any) -> int:
+        return hash(key) % self.size
+
+    def _rehash(self) -> None:
+        old_buckets = self.buckets
+        self.size *= 2
+        self.buckets = [[] for _ in range(self.size)]
+        self.count = 0
+        for bucket in old_buckets:
+            for key, value in bucket:
+                self[key] = value
```

Comment:
When rehashing, using `self[key] = value` is correct, but ensure that the `__setitem__` method handles rehashing properly. It seems to be correct here, but double-check that no infinite loops or incorrect assignments occur during rehashing.